### PR TITLE
PLAZA-81 Refactor activating accounts

### PR DESCRIPTION
**Pull request by KevinTamCS**

### CODE REVIEW

File: backend/src/main/java/org/hackbrooklyn/plaza/security/Roles.java [hunk 1]
```diff
@@ -44,39 +49,50 @@
...
+        List<String> volunteerAuthorities = Arrays.asList(
+                // TODO: Add volunteer authorities
+        );
 
         // Participants
-        String[] participantAuthorities = {
+        List<String> participantAuthorities = Arrays.asList(
                 // Announcements
-                Authorities.ANNOUNCEMENTS_READ,
+                Authorities.ANNOUNCEMENTS_READ_PARTICIPANTS_ONLY,
```
Comment (derickfan) [inline backend/src/main/java/org/hackbrooklyn/plaza/security/Roles.java:66]: Would Participants not be able to see the public announcements once they are signed in?

File: backend/src/main/java/org/hackbrooklyn/plaza/security/Roles.java [hunk 2]
```diff
@@ -5,34 +5,39 @@
...
-                // Applications
-                Authorities.APPLICATIONS_READ,
-                Authorities.APPLICATIONS_UPDATE_DECISION,
-                Authorities.APPLICATIONS_DELETE,
-
+        List<String> adminAuthorities = Arrays.asList(
                 // Announcements
                 Authorities.ANNOUNCEMENTS_CREATE,
-                Authorities.ANNOUNCEMENTS_READ,
+                Authorities.ANNOUNCEMENTS_READ_PUBLIC,
```
Comment (derickfan) [inline backend/src/main/java/org/hackbrooklyn/plaza/security/Roles.java:33]: Shouldn't Admin also be given the ANNOUNCEMENTS_READ_PARTICIPANTS_ONLY authority

**Review by derickfan — COMMENTED** [review]
Just some questions about ANNOUNCEMENTS_READ_PUBLIC amd ANNOUNCEMENTS_READ_PARTICIPANTS_ONLY authorities

File: backend/src/main/java/org/hackbrooklyn/plaza/security/Roles.java (hunk 1 above)
Comment (KevinTamCS) [reply backend/src/main/java/org/hackbrooklyn/plaza/security/Roles.java:66]: Participants would in fact be able to see public announcements. Later on in the code, we combine the authorities for applicants (with public announcement access) with the authorities for participants, granting participants access to public announcements.

File: backend/src/main/java/org/hackbrooklyn/plaza/security/Roles.java (hunk 2 above)
Comment (KevinTamCS) [reply backend/src/main/java/org/hackbrooklyn/plaza/security/Roles.java:33]: Yep! Latest push should fix this issue.

**Review by derickfan — APPROVED** [review]
LGTM

**Pull request merged**